spec/records/delete_record_spec.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+require_relative '../helpers/spec_helper'
+require_relative '../helpers/auth_helper'
+require_relative '../helpers/logs_helper'
+require_relative '../../app/models/records'
+require_relative '../helpers/status_codes_shared_examples'
+require_relative '../../app/controllers/sessions_controller'
+require_relative '../../app/controllers/records_controller'
+require_relative '../../app/controllers/users_controller'
+require_relative '../../app/controllers/application_controller'
+require 'yaml'
+
+data = YAML.load_file('data.yml')
+
+RSpec.describe '[Records API, delete record]' do
+  include AuthHelper
+  include LoggerHelper
+
+  context 'when GET /records/:id/delete' do
+
+    let(:random_fn) { Faker::Name.name }
+    let(:random_sn) { Faker::Name.last_name }
+    let(:record_access_error) { data['record_access_error'] }
+
+    context 'when not authorised, with valid record id' do
+      record = FactoryBot.create(:records)
```

Comment:
Is it okay to define record just within the context block? If I wrap it with 'let' or move to 'before' section, it will not be accessible within 'it' or 'it_behaves_like'

Reply:
Defining record within the context block as shown in your code is a valid approach if you only need access to it within that specific context. It will be accessible within that context block and any nested it blocks or other blocks within that context.

However, if you need to use the record variable across multiple context blocks or it blocks within the same test suite, you may want to define it in the before block or using let outside of any specific context block to make it accessible globally within the test suite. This would allow you to reuse the record variable throughout your tests.

Reply:
I mean: let(:record) { FactoryBot.create(:record) } create before context 'when GET /records/:id/delete' do